---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# cellpanelr

<!-- badges: start -->
[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
<!-- badges: end -->

Identify predictive biomarkers from cell line panels. Correlate your data with mutations, gene expression, and more. The goal of cellpanelr is to make "omics" level data analysis accessible and open-source for everyone.

cellpanelr uses data sets adapted from [DepMap (Broad Institute)](<https://depmap.org/portal/>) under the [CC BY 4.0](<https://creativecommons.org/licenses/by/4.0/>) license.

## Web application

The interactive analysis tool is available at <https://dwassarman.shinyapps.io/cellpanelr/>

## Package installation

To install cellpanelr from GitHub, enter the following command in an interactive R session

```{r, eval = FALSE}
# install.packages("remotes")
remotes::install_github("dwassarman/cellpanelr")
```

- Note: if you get the following error `Error in loadNamespace(x) : there is no package called ‘remotes’`. Remove the `#` character from the command above to install the `remotes` package.

## Usage

- `run_app()` runs a local instance of the interactive shiny app
- `add_ids()` matches cell line names with DepMap IDs for subsequent analysis
- `cor_expression()` and `cor_mutations()` correlate cell line response data with gene expression and gene mutations
- `data_annotations()`, `data_expression()`, and `data_mutations()` retrieve modified DepMap data sets for over 1,000 cell lines
- `data_nutlin()` provides an example data set containing cell line sensitivity to the drug nutlin-3. See the example below for a more detailed analysis of this data set.

```{r}
library(cellpanelr)
library(tidyverse) # Read in data, data joining, pipe operator
```
```{r, eval=FALSE}
# Load data
data <- read_csv("cell_viability.csv")

# Add depmap_id column
# Example: the "Cell line" column contains the cell line names
data <- add_ids(data, cell_col = "Cell line")

# Add cell line annotations for each cell line
annotated <- data %>%
  left_join(
    data_annotations(),
    by = "depmap_id"
  )

# Correlate response column with gene expression
# Example: response values are in the "viability" column
exp_results <- cor_expression(data, response = "viability", ids = "depmap_ids")

# Correlate response column with mutations
# Example: response values are in the "viability" column
mut_results <- cor_mutations(data, response = "viability", ids = "depmap_ids")
```

## Example

Using the nutlin-3 example data set, we can identify biomarkers that predict sensitivity to the drug nutlin-3.

```{r load data}
# Get the nutlin-3 sensitivity data
nutlin <- data_nutlin()

# Take a look at this tibble
glimpse(nutlin)
```

```{r prep data}
# Clean data for next steps
nutlin_clean <-
  nutlin %>%
  # Add DepMap IDs
  add_ids(cell = "Cell line") %>%
  # Remove cell lines that weren't matched to an ID
  filter(!is.na(depmap_id)) %>%
  # Remove cell lines without AUC values
  filter(!is.na(AUC))

glimpse(nutlin_clean)
```

```{r expression}
# Correlate gene expression with nutlin-3 AUC
exp_result <- 
  nutlin_clean %>%
  cor_expression(
    response = "AUC",
    ids = "depmap_id"
    )

glimpse(exp_result)

# Merge expression correlations with input nutlin-3 data
exp_merged <- 
  nutlin_clean %>%
  inner_join(data_expression(), by = "depmap_id") %>%
  left_join(exp_result, by = "gene")

glimpse(exp_merged)
```

```{r mutation}
# Correlate gene expression with nutlin-3 AUC
mut_result <- 
  nutlin_clean %>%
  cor_mutations(
    response = "AUC",
    ids = "depmap_id"
  )

glimpse(mut_result)

# Merge expression correlations with input nutlin-3 data
mut_merged <- 
  nutlin_clean %>%
  inner_join(data_mutations(), by = "depmap_id") %>%
  left_join(mut_result, by = "gene")

glimpse(mut_merged)
```

```{r plotting}
# Plot top gene expression biomarkers
exp_merged %>%
  # Filter for 12 genes with strongest negative correlation
  filter(dense_rank(rho) <= 12) %>%
  # Convert gene to a factor so that faceted plot is sorted by rho
  mutate(gene = fct_reorder(gene, rho)) %>%
  # Create plot
  ggplot(aes(x = rna_expression, y = AUC)) +
  geom_point(alpha = 0.2) +
  geom_smooth(method = "lm", se = FALSE) +
  xlab("RNA expression (log2[TPM + 1])") +
  ylab("Area under curve") +
  ggtitle("Top sensitizing genes") +
  facet_wrap(~gene, scales = "free") +
  theme_bw()

# Plot top mutation biomarkers
mut_merged %>%
  # Plot only significant mutations
  filter(significant) %>%
  # Convert gene to a factor so that faceted plot is sorted by p.value
  mutate(gene = fct_reorder(gene, p.value)) %>%
  # Change naming of mutant to be more explicit
  mutate(mutant = ifelse(mutant, "Mutant", "Wild-type")) %>%
  mutate(mutant = factor(mutant, levels = c("Wild-type", "Mutant"))) %>%
  # Create plot
  ggplot(aes(x = mutant, y = AUC)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = mutant), width = 0.2, alpha = 0.4) +
  facet_wrap(~gene) +
  scale_color_viridis_d(option = "C", end = 0.8) +
  theme_bw() +
  theme(legend.position = "none") +
  xlab("Genotype") +
  ylab("Area under curve") +
  ggtitle("Significant mutation biomarkers")

```


## Code of Conduct

Please note that the cellpanelr project is released with a [Contributor Code of Conduct](https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html). By contributing to this project, you agree to abide by its terms.
